### PR TITLE
Fix for broken relevant search in issue 359

### PR DESCRIPTION
**Pull request by PaulAnnekov**
I have added the Relevanssi plugin to mu-plugins folder and added its initialization during a site installation phase. Try it.

Issue: [Search at api.jquery.com gives inaccurate results](https://github.com/jquery/jquery-wp-content/issues/359).

### CODE REVIEW

**Comment by scottgonzalez** [comment]
So this requires re-installing all the sites in order for searching to work?

**Comment by PaulAnnekov** [comment]
We have multiple options:

1. If you have some sort of one-time tasks then we can set up Relevanssi there.
2. You can launch a piece of php code that I can write to set up Relevannsi on live server.
3. We can add some special flag (maybe using get_option/update_option) that indicates that plugin was not set up and launch setup depending on it.

**Comment by scottgonzalez** [comment]
@jquery/infrastructure How do you want to handle this?

**Comment by PaulAnnekov** [comment]
Any progress?

**Comment by scottgonzalez** [comment]
I'm really not sure what we should do. I feel like whatever solution we use has to Just Work, otherwise everyone will end up with broken sites.

@nacin thoughts?

**Comment by PaulAnnekov** [comment]
Ok, as I understand I need to implement third approach:
> We can add some special flag (maybe using get_option/update_option) that indicates that plugin was not set up and launch setup depending on it.

We will check some flag on each request and update Relevanssi settings if they are not set up yet.

**Comment by scottgonzalez** [comment]
Ok, let's try that out.

**Comment by PaulAnnekov** [comment]
I took the time and moved Relevanssi installation from blog install to blog init (on each user request). Now it will "Just Work" :).

**Comment by PaulAnnekov** [comment]
hey, guys, let's close this bug

**Comment by arthurvr** [comment]
I'm still interested in hearing @nacin's thoughts on this.

**Comment by PaulAnnekov** [comment]
@arthurvr so will he say anything? :)

**Comment by deltab** [comment]
Still waiting? :-)

**Comment by PaulAnnekov** [comment]
lol, yes :)

**Comment by mgol** [comment]
@scottgonzalez @gnarf @Krinkle who would be best to review this?

**Comment by Krinkle** [comment]
I can give this a shot, although I don't have rights in this specific repo at the moment.

@PaulAnnekov Can you clarify how `mu-plugins/relevanssi-install.php` is meant to be loaded? From a quick glance it's unclear to me whether it gets loaded at all, and if so, what ensures it will load *after* the main plugin.

The previous way of calling a function explicitly was easier to follow. I've been working on various WordPress sites, plugins, and themes for years, but don't have much experience with MU/Multisite yet (aside from the occasional fix on jQuery sites).

I thought perhaps `-install.php` is a special pattern in our custom code or in WordPress, but that doesn't seem to be the case. Am I correct in assuming you're essentially registering `relevanssi-install.php` as its own plugin? And it is not loaded via `_loader.php`, but rather loaded directly by WordPress itself (given _loader only considers specific sub directories)? If so, then I assume the idea is that WordPress loads first `_loader.php` (including the main plugin), and then later loads `relevanssi-install.php`. That should work, so long WordPress uses sorted load order and not the default `opendir()` order (Edit: [confirmed](https://github.com/WordPress/WordPress/blob/6fda2e67b0fe3872cbf5f82b58b98f2a4c96d8f8/wp-includes/load.php#L566-L580), sorts before inclusion)

File: mu-plugins/_loader.php [hunk 1]
```diff
@@ -14,4 +14,5 @@
 }
 unset( $live_domain, $subdomain, $domain_specific_file, $type_specific_file );
 
+require_once(WPMU_PLUGIN_DIR . '/relevanssi/relevanssi.php');
```
Comment (Krinkle) [inline mu-plugins/_loader.php]: Avoid using `require_once` like a function with parenthesis. Instead, use it as a statement like the  line below.

**Comment by PaulAnnekov** [comment]
@Krinkle this article will answer ALL your questions https://codex.wordpress.org/Must_Use_Plugins. WordPress runs only directly placed in `mu-plugins` files. In alphabetical order. The whole process is like this:
1. `_loader.php` loaded. It includes relevanssi plugin. As article above says, no plugin hooks will run.
2. `relevanssi-install.php` is running. It checks if it's not root blog and `jquery_relevanssi_installed` flag is not set. If it's true, it's starting installation process, setting custom config and `jquery_relevanssi_installed` flag to do not repeat this process again.

That's it.

**Comment by mgol** [comment]
I rebased & squashed this PR and re-submitted as #408.

I'd like to land it but I don't have enough WordPress+PHP expertise to review it thouroughly.

@Krinkle are you confident it's OK to land #408? If so, I'll do it but I'd like some kind of confirmation.

**Comment by Krinkle** [comment]
@mgol Aye, yeah, let's close this pull in favour of https://github.com/jquery/jquery-wp-content/pull/408.

**Pull request closed without merging**